builtin/common/settings/dlg_change_mapgen_flags.lua

```diff
@@ -46,33 +46,54 @@ local function get_formspec(dialogdata)
 	end
 
 	local fields = {}
-	local function add_field(x, name, label, value)
+	local function add_field(x, name, label, value, tooltips)
 		fields[#fields + 1] = ("field[%f,%f;3.3,1;%s;%s;%s]"):format(
 			x, height, name, label, core.formspec_escape(value or "")
 		)
+		if tooltips then
+			fields[#fields + 1] = ("tooltip[%s;%s]"):format(name, table.concat(tooltips, "\n"))
+		end
 	end
 	-- First row
 	height = height + 0.3
-	add_field(0.3, "te_offset", fgettext("Offset"), t[1])
-	add_field(3.6, "te_scale",  fgettext("Scale"),  t[2])
-	add_field(6.9, "te_seed",   fgettext("Seed"),   t[6])
+	add_field(0.3, "te_offset", fgettext("Offset"), t[1], {fgettext("This value offsets the final noise.")})
+	add_field(3.6, "te_scale",  fgettext("Scale"),  t[2], {
+			fgettext("This value is multiplied with the noise before the offset is added.")})
+	add_field(6.9, "te_seed",   fgettext("Seed"),   t[6], {
+			fgettext("This value acts as a random seed for the noise. The same seed results in the same noise.")})
 	height = height + 1.1
 
 	-- Second row
-	add_field(0.3, "te_spreadx", fgettext("X spread"), t[3])
+	add_field(0.3, "te_spreadx", fgettext("X spread"), t[3], {
+			fgettext("This value scales the noise in the X axis by this value."),
+			fgettext("This is also the scale of the largest structures in the X direction of the noise.")})
 	if dimension == 3 then
-		add_field(3.6, "te_spready", fgettext("Y spread"), t[4])
+		add_field(3.6, "te_spready", fgettext("Y spread"), t[4], {
+			fgettext("This value scales the noise in the Y axis by this value."),
+			fgettext("This is also the scale of the largest structures in the Y direction of the noise.")})
 	else
 		fields[#fields + 1] = "label[4," .. height - 0.2 .. ";" ..
 				fgettext("2D Noise") .. "]"
 	end
-	add_field(6.9, "te_spreadz", fgettext("Z spread"), t[5])
+	add_field(6.9, "te_spreadz", fgettext("Z spread"), t[5], {
+			fgettext("This value scales the noise in the Z axis by this value."),
+			fgettext("This is also the scale of the largest structures in the Z direction of the noise.")})
 	height = height + 1.1
 
 	-- Third row
-	add_field(0.3, "te_octaves", fgettext("Octaves"),     t[7])
-	add_field(3.6, "te_persist", fgettext("Persistence"), t[8])
-	add_field(6.9, "te_lacun",   fgettext("Lacunarity"),  t[9])
+	add_field(0.3, "te_octaves", fgettext("Octaves"),     t[7], {
+			fgettext("This value controls how many octaves the noise will have."), "",
+			fgettext("An octave is a simple noise generator."),
+			fgettext("Its scale and amplitude will be multiplied based on the Persistence and Lacunarity settings."),
+			fgettext("Finally, all of the octaves will be added to generate this noise.")})
```

Comment:
IMO too complicated. I'd suggest to shorten this to "The higher the octave count, the more fine-grained or detailed noise becomes. This in turn also costs computing time.".

I'd greatly welcome similar simplifications for persistence and lacunarity.